src/backend/columnar/columnar_reader.c

```diff
@@ -283,12 +301,27 @@ BeginStripeRead(StripeMetadata *stripeMetadata, Relation rel, TupleDesc tupleDes
 
 
 /*
- * EndStripeRead finishes a stripe read.
+ * EndStripeRead finishes current stripe read.
+ */
+static void
+EndStripeRead(ColumnarReadState *readState)
```

Comment:
The destructor should just take a pointer to the thing being destructed, the stripeReadState. The memory context can be reset when advancing.